src/Helper.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\unl_cas;
+use Drupal\user\Entity\User;
+use Drupal\user\UserDataInterface;
+
+/**
+ * Configures unl_cas settings for this site.
+ */
+class Helper
+{
+
+  function __construct()
+  {
+    //nothing to do here
+  }
+
+  /**
+   * Import a UNL user into the system and return the drupal User object
+   *
+   * @param $username
+   *
+   * @return bool|\Drupal\Core\Entity\EntityInterface|object|static
+   */
+  public function importUser($username) {
+    $username = trim($username);
+    $user = user_load_by_name($username);
+    if (!$user) {
+      $user = User::create();
+      $user->setUsername($username);
+      $user->setEmail($username . '@unl.edu');
+      $user->enforceIsNew();
+      $user->activate();
+    }
+    
+    $user->save();
+    
+    //The first time that they log in, try to update userdata
+    $this->updateUserData($user);
```

Comment:
This is going to run every time a user logs in.  If the idea is to only have this run on new account creation, and let unl_cas_user_login() handle adding to the queue on subsequent logins, shouldn't this be up inside the "if (!$user) {}" block on line 28?